src/TwigComponent/src/ComponentRenderer.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TwigComponent;
+
+use Twig\Environment;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @experimental
+ */
+final class ComponentRenderer
+{
+    private $twig;
+
+    public function __construct(Environment $twig)
+    {
+        $this->twig = $twig;
+    }
+
+    public function render(ComponentInterface $component): string
+    {
+        // TODO: Template attribute/annotation/interface to customize
+        // TODO: Self-Rendering components?
+        $templateName = sprintf('components/%s.html.twig', $component::getComponentName());
```

Comment:
maybe let this configurable?  or call it twig_components? `components` is a generic name

Reply:
We may actually move the entire template name into the component itself - e.g. via a `getTemplate(): string` method. It seems like people are liking that better.

Reply:
it would also make it compatible with bundles providing some components, btw